packages/runtime/src/errors.ts

```diff
@@ -137,9 +182,34 @@ export class AdaptorError extends RTError {
   severity = 'fail';
   message: string = '';
   details: any;
-  constructor(error: any) {
+
+  line?: number;
+  operationName?: string;
+
+  constructor(error: any, line?: number, operationName?: string) {
     super();
-    this.details = error;
+    if (!isNaN(line!)) {
```

Comment:
Quite tricky. `isNaN` doesn't do great job of telling whether a value isn't a number. 

It thinks:
1. an empty string is a number
2. `null` is a number

Hence, using `!isNaN` to check for something being a number might be flawed when the value passed at runtime is either `null` or `''`. 

Great to add a `typeof` check to make it get only numbers. 
```suggestion
    if (typeof line === "number" && !isNaN(line!)) {
```

Hence, typeof == number will make sure that we're scoped to numbers(1,2,3,.... & NaN).

Reply:
In case a function like `parseInt` which is supposed to always return a number(NaN if not) is used before here, then line will always be in the number scope.

Reply:
Is `typeof number` robust? For some reason I didn't think it was :thinking:  Maybe I've invented that...

Reply:
Oh look, lodash uses `typeof number`. OK cool, I accept!

Reply:
Ok so I look at this code and I had no idea what was going on :shrug: 

The number thing isn't important. Basically when generating an adaptor error, we EITHER attribute it to a source offset in the VM, OR we attribute it to an operation (with a name and line number)

In https://github.com/OpenFn/kit/pull/848/commits/8882c4cd984b012dc135f88a45df14585de59662 I've cleaned up the AdaptorError so that the constructor reflects the correct behaviour. I've dropped a couple of unit tests too to explain and enforce the behaviour. Although the real intereesting logic behind this behaviour is elsewhere.